source/features/useful-not-found-page.tsx

```diff
@@ -28,119 +25,360 @@ async function checkAnchor(anchor: HTMLAnchorElement): Promise<void> {
 }
 
 function parseCurrentURL(): string[] {
-	const parts = getCleanPathname().split('/');
-	if (parts[2] === 'blob') { // Blob URLs are never useful
-		parts[2] = 'tree';
+	const parts = getCleanPathname().split("/");
+	// Blob URLs are never useful
+	if (parts[2] === "blob") {
+		parts[2] = "tree";
 	}
 
 	return parts;
 }
 
-// If the resource was deleted, link to the commit history
-async function addCommitHistoryLink(bar: Element): Promise<void> {
+async function getUrlToFileOnDefaultBranch(): Promise<string | undefined> {
+	// Get the current branch
 	const parts = parseCurrentURL();
-	parts[2] = 'commits';
-	const url = '/' + parts.join('/');
+	const currentBranch = parts[3];
+
+	if (!currentBranch) {
+		return;
+	}
+
+	const defaultBranch = await getDefaultBranch();
+	if (currentBranch === defaultBranch) {
+		return;
+	}
+
+	// Change branch
+	parts[3] = defaultBranch;
+	const url = "/" + parts.join("/");
+	// Check if that path exists
 	if (await is404(location.origin + url)) {
 		return;
 	}
 
-	bar.after(
-		<p className="container mt-4 text-center">
-			See also the {getType()}’s <a href={url}>commit history</a>.
-		</p>,
+	// If it does, return it
+	return url;
+}
+
+async function getLastCommitForFile(
+	branch?: string
+): Promise<Record<string, string | undefined> | undefined> {
+	// Get the file path from the parts
+	const parts = parseCurrentURL();
+	let filePath = parts.slice(4).join("/");
```

Comment:
Refer to `GitHubURL` to parse the URL, I think it works here as well (I might be wrong)